clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -11049,7 +11049,11 @@ void LinkerWrapper::ConstructJob(Compilation &C, const JobAction &JA,
     SYCLInstallationDetector SYCLInstallation(D);
     SYCLInstallation.getSYCLDeviceLibPath(LibLocCandidates);
     SmallString<128> LibName("libsycl-crt");
-    StringRef LibSuffix = TheTriple.isWindowsMSVCEnvironment() ? ".obj" : ".o";
+    bool IsNewOffload = Args.hasFlag(options::OPT_offload_new_driver,
```

Comment:
Instead of checking for the args each time - you can use `D.getUseNewOffloadingDriver()`

Reply:
**NIT**:  Possibly not directly related to this PR, but `getUseNewOffloadingDriver()` could be renamed to `getNewOffloadingDriver()` or `useNewOffloadingDriver()` . `getUse*` sounds a bit odd to me.

Reply:
Here, we are not reading in a driver. We are reading in a boolean variable called UseNewOffloadingDriver. I suppose the naming seems ok. This is an upstream function though. If we have to change the name, we will have to do it upstream. Will keep this in mind for a future update. Hope that's ok. Thanks